### PR TITLE
Ground Truth Uploading

### PR DESCRIPTION
**Pull request by RobGeada**
Added flag to enable data/upload endpoint to receive ground truth data. This will create/access a ground truth dataframe. Datapoint ids will mandatorily correlate across the recorded data and ground truth dataframes, allowing for the retrieval of correlated ground truths. 

Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `FAI-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: explainability-service/src/test/java/org/kie/trustyai/service/endpoints/data/DataEndpointTest.java [hunk 1]
```diff
@@ -439,4 +444,118 @@ void uploadTagThatUsesProtectedName() {
...
+    }
+
+    @Test
+    void uploadDataAndGroundTruth() {
+        int[] testInputRows = new int[] { 1, 5, 250 };
+        int[] testInputCols = new int[] { 1, 4 };
+        int[] testOutputCols = new int[] { 1, 2 };
+        String[] testDatatypes = new String[] { "INT64", "INT32", "FP32", "FP64", "BOOL" };
+
+        // sorry for the quad loop
```
Comment (ruivieira) [inline explainability-service/src/test/java/org/kie/trustyai/service/endpoints/data/DataEndpointTest.java:467]: @RobGeada could you create a ticket to optimise this, and replace the comment with the issue link? Thanks.

**Review by ruivieira — CHANGES_REQUESTED** [review]

File: explainability-core/src/main/java/org/kie/trustyai/explainability/model/Dataframe.java [hunk 2]
```diff
@@ -46,23 +47,28 @@ public class Dataframe {
...
         this.internalData = new InternalData();
+        this.idToIDX = new HashMap<>();
     }
 
     Dataframe(List<List<Value>> data, Metadata metadata, InternalData internalData) {
         this.data = new ArrayList<>(data);
         this.metadata = metadata;
         this.internalData = internalData;
+        this.idToIDX = new HashMap<>();
+        calculateIndexHashes();
```
Comment (danielezonca) [inline explainability-core/src/main/java/org/kie/trustyai/explainability/model/Dataframe.java]: Minor comment (this is going to always trigger the `calculateIndexHashes()` method but it is going to be a no-op when `internalData` is empty)
```suggestion
    public Dataframe() {
        this(new ArrayList<>(new ArrayList<>()), new Metadata(), new InternalData());
    }

    Dataframe(List<List<Value>> data, Metadata metadata) {
        this(new ArrayList<>(data), metadata, new InternalData());
    }

    Dataframe(List<List<Value>> data, Metadata metadata, InternalData internalData) {
        this.data = new ArrayList<>(data);
        this.metadata = metadata;
        this.internalData = internalData;
        this.idToIDX = new HashMap<>();
        calculateIndexHashes();
```

**Review by danielezonca — APPROVED** [review]

**Review by ruivieira — APPROVED** [review]

**Pull request merged**